connectors/storages.mongodb/src/storage.js

```diff
@@ -100,13 +98,34 @@ class Storage extends Connector {
       $inc: { _version: 1 }
     }
 
-    options.returnDocument = 'after'
+    options.returnDocument = ReturnDocument.AFTER
 
-    const result = await this.#connection.update(criteria, update, options)
+    const result = await this.#collection.update(criteria, update, options)
 
     return from(result)
   }
 
+  async ensure (query, properties, state) {
+    let { criteria, options } = translate(query)
+
+    if (query === undefined)
+      criteria = properties
+
+    options.upsert = true
+    options.returnDocument = ReturnDocument.AFTER
+
+    const update = {
+      $setOnInsert: to(state)
+    }
+
+    const result = await this.#collection.update(criteria, update, options)
```

Comment:
NIT: not directly related to this PR but long term the semantic for this intermediatte wrapper that provides `.get` / `.update` method - while simplicity here is good is also very confusing.

For example, MongoDB has four methods that poterntially can map to `.update` - `updateOne` / `updateMany` and `findOneAndUpdate` / `findAllAndUpdate`. The semantic difference for calls on Mongo part as meaningful (and may also explain your confusion when you seen that I'm _using update to get item_) - the `.update*` method doesn't return document while `.findAndUpdate` do return document.

So, for me to contribute here a single `.update` will be a stop gap simplification as I must stop and explore:
1. Will it update just one or all?
2. Will it return document and which one - before or after update?

Overall, we should think if we need this interminent wrapper at all.

Reply:
https://github.com/toa-io/toa/pull/623/commits/190934b1cd868ddc9cd76f35ee555d1a311f0abb